### DISCUSSION TITLE
Curry

### DISCUSSION THREAD

**Discussion by Nailgy** [HowProgrammingWorks/LiveQA#14] [Q&A]
Вопрос касательно лекции по каррированию, а именно примера кода с плохим каррированием, где вместо bind используем принцип функции partial из прошлого примера под номером 5. 
Я понял, что fn.length на втором частичном применение становится 0, соответственно выполняется второе условие тернарного оператора. В итоге возвращается либо нормальное значение функции sum4, если было суммарно передано достаточно аргументов, либо значение NaN, если аргументов в итоге недостаточно. 
Можете подробнее объяснить, почему длина fn обнуляется? И ещё одно - можно ли пофиксить каррирование с принципом partial, или же этого сделать нельзя? А то переделать curry, чтобы использовался байнд несложно

**Comment by tshemsedinov** [comment]
Пофиксить можно через partial из предыдущего задания, но советую переписать это в имреративном стиле со счетчиком и более развернуто на if-ах, а не на таком огромном выражении и станет все понятнее, а потом уже исправленную свернуть в выражение
